Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstanding.cs

```diff
@@ -0,0 +1,244 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using System.Collections;
+using HoloToolkit.Unity;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// The SpatialUnderstanding class controls the state and flow of the 
+    /// scanning process used in the understanding module. 
+    /// </summary>
+    [RequireComponent(typeof(SpatialUnderstandingSourceMesh))]
+    [RequireComponent(typeof(SpatialUnderstandingCustomMesh))]
+    public class SpatialUnderstanding : Singleton<SpatialUnderstanding>
+    {
+        // Consts
+        public const float ScanSearchDistance = 8.0f;
+
+        // Enums
+        public enum ScanStates
+        {
+            None,
+            ReadyToScan,
+            Scanning,
+            Finishing,
+            Done
+        }
```

Comment:
Unless there is a very specific design reason to do so, enums should defined directly inside a namespace and not nested within a class. Otherwise, any use of the enum outside of the containing class requires prefixing the class name to use it. For example, notice the required prefix usage in SpatialUnderstandingCustomMesh line 151.